File: 프로그래머스/올바른 괄호.py
def solution(s):
    count = 0
    
    for _ in s:
        if count < 0:
            return False
        
        if _ == "(":
            count += 1
        else:
            count -= 1
    
    if (count == 0):
        return True
    else:
        return False

File: 프로그래머스/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        char[] arr = s.toCharArray();
        for (char a : arr) {
            if (a == ')' && !stack.isEmpty() && stack.peek() == '(')
                stack.pop();
            else
                stack.push(a);
        }
        return (stack.size() != 0) ? false : true;
    }
}